### PR TITLE
fixed date format to prepend a zero if the hours are less than 10

### PR DESCRIPTION
**Pull request by Meleeman01**
ref https://github.com/playasoft/volunteers/issues/132 , https://github.com/playasoft/volunteers/pull/131 , https://github.com/playasoft/volunteers/issues/68
so really the JS validation messing up was only a symptom of a bigger problem, the formatting on the backend. basically it didn't format hours properly if they were less than 10. so i added a couple ternary operators to check for that, and I left duration alone as it didn't have an affect on the app. a quick grep showed that only the schedule and schedule controller was using it.

### CODE REVIEW

File: laravel/app/Models/Schedule.php [hunk 1]
```diff
@@ -154,8 +154,12 @@ public function formatTimes()
         $end = date_parse_from_format('H:i', $this->end_time);
         $duration = date_parse_from_format('H:i', $this->duration);
 
+        ($start['hour']<10) ? $this->start_time = '0'.$start['hour'] . ":" . str_pad($start['minute'], 2, 0, STR_PAD_LEFT) :
```
Comment (itsrachelfish) [inline laravel/app/Models/Schedule.php]: These ternaries seem unnecessary. You should try using `str_pad` instead

**Comment by se7enmilgram** [comment]
As a word of warning, this isn't far off from the fix I looped into my previous commit before backing it out. The blanket under ten clause will also clobber down presets to custom times, as the prepended zero will prevent them from matching the value in the select:

The following has some indentation errors, but you can see the logic where I exempted hours 6 and 9 from formatting the start hours and 3 and 6 from the duration. The comment block above the private functions break down the reasons pretty well. 

https://github.com/se7enmilgram/volunteers/commit/00c4434d1f090473d9a6aab5476e97ccaf728b7b

**Comment by Meleeman01** [comment]
> As a word of warning, this isn't far off from the fix I looped into my previous PR before backing it out at Rachel's request. The blanket under ten clause will also clobber down presets to custom times, as the prepended zero will prevent them from matching the value in the select. This trades a UI error requiring custom times to be re-entered for a cosmetic UI bug where preset values become custom times when you edit them.
> 
> The following has some indentation errors, but you can see the logic where I exempted hours 6 and 9 from formatting the start hours and 3 and 6 from the duration. The comment block above the private functions break down the reasons pretty well.
> 
> [se7enmilgram@00c4434](https://github.com/se7enmilgram/volunteers/commit/00c4434d1f090473d9a6aab5476e97ccaf728b7b)

I see, then why aren't we using this? thats exactly what I would have done if i had seen those issues

**Comment by itsrachelfish** [comment]
> I see, then why aren't we using this? thats exactly what I would have done if i had seen those issues

This wasn't done in the previous PR because it was unrelated to the original bug being fixed. See #131 and #130 for more details

**Comment by se7enmilgram** [comment]
At the time I believed that this wasn't an issue to be handled in PHP code, but at the JS level.

File: laravel/app/Models/Schedule.php [hunk 2]
```diff
@@ -147,15 +147,23 @@ static public function setTimes($input)
...
         $end = date_parse_from_format('H:i', $this->end_time);
         $duration = date_parse_from_format('H:i', $this->duration);
 
+        //check for any default start hours and if there are, don't pad start hours
+        (!in_array($start['hour'], [0,6,9,12,24])) ?
+        $this->start_time = str_pad($start['hour'], 2, 0, STR_PAD_LEFT) . ":" . str_pad($start['minute'], 2, 0, STR_PAD_LEFT) :
         $this->start_time = $start['hour'] . ":" . str_pad($start['minute'], 2, 0, STR_PAD_LEFT);
-        $this->end_time = $end['hour'] . ":" . str_pad($end['minute'], 2, 0, STR_PAD_LEFT);
+        
+        //check for any default duration hours and if there are, don't pad duration hours
```
Comment (itsrachelfish) [inline laravel/app/Models/Schedule.php]: Why would you do this? Can't you just fix the front-end to check for padded zeroes?

File: laravel/resources/js/ui/schedule.js [hunk 3]
```diff
@@ -67,6 +67,19 @@ $(document).ready(function()
         }
     });
 
+    //format duration fields on the fly when the server can't
+    $('#duration-field').on('change', function()
+    {
+        if ($(this).value() == 'custom')
+        {
+            var currentVal = $('input[name=custom_duration]').value();
+            if(currentVal.length<5)
```
Comment (itsrachelfish) [inline laravel/resources/js/ui/schedule.js]: All of this logic (line 76-79) can be replaced by https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

File: laravel/app/Models/Schedule.php (hunk 2 above)
Comment (itsrachelfish) [reply laravel/app/Models/Schedule.php]: Simply edit the template file - https://github.com/playasoft/volunteers/blob/master/laravel/resources/views/pages/schedule/edit.blade.php#L120-L121

File: laravel/resources/js/ui/schedule.js (hunk 3 above)
Comment (se7enmilgram) [reply laravel/resources/js/ui/schedule.js]: And just to clarify, I wasn't advocating managing this in PHP, I linked my previous commit to show the backend behavior that remediated the issue. JS is almost certainly the right place to fix this, I'm just super rusty so I didn't undertake it.

**Pull request merged**